app/src/main/kotlin/com/wire/android/ui/settings/devices/DeviceDetailsScreen.kt

```diff
@@ -162,12 +206,171 @@ fun DeviceDetailsContent(
     }
 }
 
+@Composable
+private fun DeviceIdItem(state: DeviceDetailsState, onCopy: (String) -> Unit) {
+    DeviceDetailSectionContent(
+        sectionTitle = stringResource(id = R.string.label_client_device_id),
+        sectionText = AnnotatedString(state.device.clientId.formatAsString()),
+        titleTrailingItem = {
+            CopyButton(
+                onCopyClicked = {
+                    state.device.clientId.formatAsString().let { id -> onCopy(id) }
+                }
+            )
+        }
+    )
+}
+
+@Composable
+fun DeviceKeyFingerprintItem(
+    clientFingerPrint: String?,
+    onCopy: (String) -> Unit
+) {
+    DeviceDetailSectionContent(
+        stringResource(id = R.string.title_device_key_fingerprint),
+        sectionText = clientFingerPrint?.formatAsFingerPrint()
+            ?: AnnotatedString(stringResource(id = R.string.label_client_key_fingerprint_not_available)),
+        enabled = clientFingerPrint != null,
+        titleTrailingItem = {
+            CopyButton(
+                onCopyClicked = {
+                    clientFingerPrint?.let { fingerprint -> onCopy(fingerprint) }
+                },
+                state = if (clientFingerPrint != null) WireButtonState.Default else WireButtonState.Disabled
+            )
+        }
+    )
+}
+
+@Composable
+fun DeviceVerificationItem(
+    state: Boolean,
+    enabled: Boolean,
+    isSelfClient: Boolean,
+    userName: String?,
+    onStatusChange: (Boolean) -> Unit,
+) {
+    @StringRes
+    val subTitle = if (state) {
+        R.string.label_client_verified
+    } else {
+        R.string.label_client_unverified
+    }
+    DeviceDetailSectionContent(
+        stringResource(id = R.string.title_device_key_fingerprint),
+        AnnotatedString(stringResource(id = subTitle)),
+        titleTrailingItem = {
+            WireSwitch(
+                checked = state,
+                onCheckedChange = onStatusChange,
+                enabled = enabled,
+                modifier = Modifier.padding(end = dimensions().spacing16x)
+            )
+        }
+    )
+    VerificationDescription(isSelfClient, userName)
+}
+
+@Composable
+private fun VerificationDescription(
+    isSelfClient: Boolean,
+    userName: String?
+) {
+    val context = LocalContext.current
+    Column(
+        modifier = Modifier
+            .fillMaxWidth()
+            .padding(
+                start = dimensions().spacing16x,
+                end = dimensions().spacing16x,
+                bottom = dimensions().spacing16x
+            )
+    ) {
+
+        if (isSelfClient) {
+            DescriptionText(
+                text = stringResource(id = R.string.label_self_client_verification_description),
+                leanMoreLink = "https://support.wire.com/hc/en-us/articles/207692235-How-can-I-compare-key-fingerprints-"
```

Comment:
Let's localize all these support links to the strings file, similar to what we did to others. There is probably a German support version of them, so ze German users will appreciate it too 🤓